sdk/spring/azure-spring-cloud-stream-binder-eventhubs/src/main/java/com/azure/spring/eventhub/stream/binder/EventHubMessageChannelBinder.java

```diff
@@ -56,9 +56,7 @@ protected MessageHandler createProducerMessageHandler(ProducerDestination destin
         handler.setSendTimeout(producerProperties.getExtension().getSendTimeout());
         handler.setSendFailureChannel(errorChannel);
         if (producerProperties.isPartitioned()) {
-            if (producerProperties.getPartitionKeyExpression() != null) {
-                handler.setPartitionKeyExpression(producerProperties.getPartitionKeyExpression());
-            }
+            handler.setPartitionIdExpressionString("headers['" + BinderHeaders.PARTITION_HEADER + "']");
```

Comment:
I think we could remove the two properties: partitionIdExpressionString and partitionKeyExpressionString, and directly parse the string to an expression here, and then pass it to the handler.